src/encoded/helpers/helper.py

```diff
@@ -0,0 +1,118 @@
+                if assembly in viz:  # mm10 and mm10-minimal resolve to the same thing
+                    continue
+                search_params = request.query_string.replace('&', ',,')
+                if not request.params.getall('assembly') \
+                or assembly in request.params.getall('assembly'):
+                    # filter  assemblies that are not selected
+                    hub_url = request.route_url('batch_hub', search_params=search_params,
+                                                txt='hub.txt')
+                    browser_urls = {}
+                    pos = None
+                    if 'region-search' in request.url and position is not None:
+                        pos = position
+                    ucsc_url = vis_format_url("ucsc", hub_url, assembly, pos)
+                    if ucsc_url is not None:
+                        browser_urls['UCSC'] = ucsc_url
+                    ensembl_url = vis_format_url("ensembl", hub_url, assembly, pos)
+                    if ensembl_url is not None:
+                        browser_urls['Ensembl'] = ensembl_url
+                    if browser_urls:
+                        viz[assembly] = browser_urls
+                        #actions.setdefault('visualize_batch', {})[assembly] =\
+                                #browser_urls  # formerly 'batch_hub'
+        if viz:
+            actions.setdefault('visualize_batch', viz)
+
+    # generate batch download URL for experiments
+    # TODO we could enable them for Datasets as well here, but not sure how well it will work
+    # batch download disabled for region-search results
+    if '/region-search/' not in request.url:
+        #if (doc_types == ['Experiment'] or doc_types == ['Annotation']) and any(
+        if (doc_types == ['Experiment']) and any(
+                bucket['doc_count'] > 0
+                for bucket in aggregations['files-file_type']['files-file_type']['buckets']):
+            actions['batch_download'] = request.route_url(
+                'batch_download',
+                search_params=request.query_string
+            )
+
+    return actions
+
+class View_Item:
+    def __init__(self, request, search_base):
+        self.result_list = {
+            'href': request.route_path('search', slash='/') + search_base,
+            'title': 'View results as list',
+            'icon': 'list-alt',
+        }
+        self.tabular_report = {
+            'href': request.route_path('report', slash='/') + search_base,
+            'title': 'View tabular report',
+            'icon': 'table',
+        }
+        self.summary_report = {
+            'href': request.route_path('summary', slash='/') + search_base,
+            'title': 'View summary report',
+            'icon': 'summary',
+        }
+        self.summary_matrix = {
+            'href': request.route_path('matrix', slash='/') + search_base,
+            'title': 'View summary matrix',
+            'icon': 'th',
+        }
```

Comment:
Can you look into how to get rid of the 'no newline' at the end of the file?  I think we want new lines.